### ISSUE TITLE
Odd build string?

### ISSUE THREAD

**Issue by dhirschfeld** [mamba-org/boa#302]
If I specify `python >=3.10` in my `host` section I'm get the build string `1.2.3-py>=310h7736c20_0`

If I instead specify `python 3.10` I get the build string `1.2.3-py310h7736c20_0` which seems a lot more sensible.

I'm not sure it makes much sense to embed the `>=` in the build string, but maybe that's a user error and you have to specify a single concrete version rather than a bound?

Additionally, if I specify `noarch: python` I get the build string `1.2.3-pypyh7736c20_0`  which makes it appear to be for the `pypy` platform - is that intended?

**Comment by wolfv** [comment]
Indeed, those build strings look odd. I would have to take a look at this again.
